app/models/episode_story_handler.rb

```diff
@@ -0,0 +1,95 @@
+require 'episode'
+
+class EpisodeStoryHandler
+  include PRXAccess
+
+  attr_accessor :episode, :story
+
+  def initialize(episode)
+    self.episode = episode
+  end
+
+  def self.create_from_story!(story)
+    series_uri = story.links['series'].href
+    podcast = Podcast.find_by!(prx_uri: series_uri)
+    episode = Episode.new(podcast: podcast)
+    update_from_story!(episode, story)
+  end
+
+  def self.update_from_story!(episode, story = nil)
+    story ||= get_story
+    new(episode).update_from_story!(story)
+  end
+
+  def update_from_story!(story)
+    Episode.transaction do
+      episode.lock!
+      update_from_story(story)
+      episode.save!
+    end
+    episode
+  end
+
+  def update_from_story(story)
+    self.story = story
+    self.episode.prx_uri = story.links['self'].href
+    self.episode.url = story.links['alternate'].try(:href)
+
+    update_attributes
+    update_audio
+  end
+
+  def update_attributes
+    sa = story.attributes
+    self.episode.title = sa[:title]
+    self.episode.subtitle = Sanitize.fragment(sa[:short_description] || '').strip
+    self.episode.description = Sanitize.fragment(sa[:description] || '').strip
+    self.episode.summary = sa[:description]
+    self.episode.content = sa[:description]
+    self.episode.categories = sa[:tags]
+    self.episode.published_at = Time.parse(sa[:published_at]) if sa[:published_at]
+    self.episode.updated = Time.parse(sa[:updated_at]) if sa[:updated_at]
+  end
+
+  def update_audio
+    audio = story.objects["prx:audio"].objects['prx:items'] rescue []
+    audio.each do |a|
+      next unless a.status == 'complete'
```

Comment:
I'm guessing this is the magic that prevents publishing an episode before the media completes processing?